### ISSUE TITLE
Security example with @EnableGlobalMethodSecurity  causes  java.io.FileNotFoundException 

### ISSUE THREAD

**Issue by efrengarcial** [spring-projects-experimental/spring-graalvm-native#289]
hi,

I have updated the "Security" example and added the annotation @EnableGlobalMethodSecurity  to the class WebSecurityConfig:

Sample Project: spring-graalvm-native-samples/security

```@Configuration(proxyBeanMethods = false)
@EnableWebSecurity
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
	@Override
	protected void configure(HttpSecurity http) throws Exception {
		http
			.authorizeRequests()
				.antMatchers("/", "/home").permitAll()
				.anyRequest().authenticated()
				.and()
			.formLogin()
				.loginPage("/login")
				.permitAll()
				.and()
			.logout()
				.permitAll();
	}

	@Bean
	@Override
	public UserDetailsService userDetailsService() {
		UserDetails user =
			 User.withDefaultPasswordEncoder()
				.username("user")
				.password("password")
				.roles("USER")
				.build();

		return new InMemoryUserDetailsManager(user);
	}
}
```

But it fails at the moment of running the application, Full Stack Trace:

```
2020-09-17 08:02:30.766  INFO 11442 --- [           main] c.e.securingweb.SecuringWebApplication   : Starting SecuringWebApplication using Java 11.0.8 on server with PID 11442 (/opt/java/apps/spring-graalvm-native/spring-graalvm-native-samples/security/target/security started by efren in /opt/java/apps/spring-graalvm-native/spring-graalvm-native-samples/security)
2020-09-17 08:02:30.767  INFO 11442 --- [           main] c.e.securingweb.SecuringWebApplication   : No active profile set, falling back to default profiles: default
2020-09-17 08:02:30.790  WARN 11442 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [com.example.securingweb.WebSecurityConfig]; nested exception is java.io.FileNotFoundException: class path resource [org/springframework/security/config/annotation/method/configuration/MethodSecurityMetadataSourceAdvisorRegistrar.class] cannot be opened because it does not exist
2020-09-17 08:02:30.793 ERROR 11442 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [com.example.securingweb.WebSecurityConfig]; nested exception is java.io.FileNotFoundException: class path resource [org/springframework/security/config/annotation/method/configuration/MethodSecurityMetadataSourceAdvisorRegistrar.class] cannot be opened because it does not exist
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:609) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:310) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:249) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:198) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.doProcessConfigurationClass(ConfigurationClassParser.java:303) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.processConfigurationClass(ConfigurationClassParser.java:249) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:206) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.parse(ConfigurationClassParser.java:174) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:336) ~[security:na]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:252) ~[security:na]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:285) ~[na:na]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:99) ~[na:na]
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:751) ~[na:na]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:569) ~[na:na]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144) ~[na:na]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759) ~[security:na]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:751) ~[security:na]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:410) ~[security:na]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:326) ~[security:na]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1280) ~[security:na]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1269) ~[security:na]
	at com.example.securingweb.SecuringWebApplication.main(SecuringWebApplication.java:10) ~[security:na]
Caused by: java.io.FileNotFoundException: class path resource [org/springframework/security/config/annotation/method/configuration/MethodSecurityMetadataSourceAdvisorRegistrar.class] cannot be opened because it does not exist
	at org.springframework.core.io.ClassPathResource.getInputStream(ClassPathResource.java:180) ~[na:na]
	at org.springframework.core.type.classreading.SimpleMetadataReader.getClassReader(SimpleMetadataReader.java:55) ~[na:na]
	at org.springframework.core.type.classreading.SimpleMetadataReader.<init>(SimpleMetadataReader.java:49) ~[na:na]
	at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:103) ~[na:na]
	at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.createMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:86) ~[na:na]
	at org.springframework.boot.type.classreading.ConcurrentReferenceCachingMetadataReaderFactory.getMetadataReader(ConcurrentReferenceCachingMetadataReaderFactory.java:73) ~[na:na]
	at org.springframework.core.type.classreading.SimpleMetadataReaderFactory.getMetadataReader(SimpleMetadataReaderFactory.java:81) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.asSourceClass(ConfigurationClassParser.java:695) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.asSourceClasses(ConfigurationClassParser.java:674) ~[na:na]
	at org.springframework.context.annotation.ConfigurationClassParser.processImports(ConfigurationClassParser.java:581) ~[na:na]
	... 21 common frames omitted

```
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment GraalVM CE 20.3.0-dev (build 11.0.8+10-jvmci-20.2-b03)
OpenJDK 64-Bit Server VM GraalVM CE 20.3.0-dev (build 11.0.8+10-jvmci-20.2-b03, mixed mode, sharing)

**Comment by aclement** [comment]
Struggling a bit with recreating this error - I don't deny it is a problem, I just can't get that far.

If I change WebSecurityConfig to only be those pair of annotations as above (and so losing the existing ` @Configuration(proxyBeanMethods = false)`:
```
@EnableWebSecurity
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
```
then it fails complaining that `proxyBeanMethods=false` is not set for WebSecurityConfig. There is no alias for proxyBeanMethods on `EnableGlobalMethodSecurity`.

```
13:50:09.835 [main] ERROR org.springframework.boot.SpringApplication - Application run failed
org.springframework.beans.factory.BeanDefinitionStoreException: @Configuration classes need to be marked as proxyBeanMethods=false. Found: [webSecurityConfig]
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.enhanceConfigurationClasses(ConfigurationClassPostProcessor.java:437)
```

If I also add in `@Configuration(proxyBeanMethods=false)` I now get two problems:
```

13:54:38.385 [main] DEBUG org.springframework.core.LocalVariableTableParameterNameDiscoverer - ASM ClassReader failed to parse class file [class org.springframework.security.config.annotation.configuration.ObjectPostProcessorConfiguration], probably due to a new Java class file version that isn't supported yet - unable to determine constructor/method parameter names
java.lang.IllegalArgumentException: Could not find class [org.springframework.beans.factory.config.AutowireCapableBeanFactory]
	at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:334)
	at org.springframework.core.LocalVariableTableParameterNameDiscoverer$LocalVariableTableVisitor.resolveExecutable(LocalVariableTableParameterNameDiscoverer.java:233)
```
but that doesn't prevent continued startup, then: 
```
org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.boot.autoconfigure.web.servlet.ServletWebServerFactoryConfiguration$EmbeddedTomcat': Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'metaDataSourceAdvisor': Could not resolve matching constructor (hint: specify index/type/name arguments for simple parameters to avoid type ambiguities)
```

working through those right now, not quite sure why you didn't see them. Maybe related to boot 2.4 m4 upgrade? hmm.

**Comment by aclement** [comment]
Fixed this with security configuration hints. Now the app starts fine for me and works, with `@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)`. I don't see the previous issues I saw or the original one (fixed by accident?)

**Comment by sdeleuze** [comment]
Based on the comments, I think this should be fixed as of 0.8.3, please reopen if that's not the case.
